Sfeir School Expert: "ReactJS: Hooks"

JonathanGeoffroy opened the pull request:
Slides & exercises for  the sfeir school expert "ReactJS: Hooks".
FYI, due date is 29/04. (end of this month).

There are some trade-offs I made, but the main goal is to add a new expert school without touching anything in Institute part.
Please, feel free to ask for any change or suggestion before this due date :)

JonathanGeoffroy committed:
Hooks: Slides

JonathanGeoffroy committed:
Hooks : demos

JonathanGeoffroy committed:
Hooks: exercices

JonathanGeoffroy committed:
[Hooks] Slides : add cleanup

JonathanGeoffroy committed:
[Hooks] Add readme

JonathanGeoffroy committed:
[Hooks] Improve exercice boilerplate

Julien-Hery commented on src/expert/hooks/README.md:
```diff
@@ -0,0 +1,61 @@
+This will start :
+
+- Parcel bundling all exercises,
+- API server at `localhost/5000/`
```
I have nothing running at localhost:5000 or localhost/5000, did you mean localhost:1337 ? `server/index.js` uses 1337 by default

Julien-Hery commented on src/expert/hooks/README.md:
```diff
@@ -0,0 +1,61 @@
+
+## React app
+
+This Sfeir school bundles all exercises in a single web application, so that you can check each exercise at `localhost:localhost:1234/expert/hooks/{ex-name}/index.html`
```
```suggestion
This Sfeir school bundles all exercises in a single web application, so that you can check each exercise at `localhost:1234/expert/hooks/{ex-name}/index.html`
```

Julien-Hery commented on docs/markdown/expert/hooks/02-hooks/03-useEffect.md:
```diff
@@ -0,0 +1,34 @@
+<!-- .slide: class=""with-code““ -->
```
this slide does not fit the screen for me, I am on a MacBook Pro mid-2015 and sadly it's not a 1980x1080 but a 1440x900
<img width="1355" alt="Capture d’écran 2021-04-20 à 12 45 35" src="https://user-images.githubusercontent.com/24412934/115383691-743a8e80-a1d6-11eb-96fe-fbe479b4abc2.png">

JonathanGeoffroy replied:
This slide has clearly to much text IMO.  
I will split it in two slides. Thanks :)

Julien-Hery commented on docs/markdown/expert/hooks/07-reducer-context/04-solution-04.md:
```diff
@@ -0,0 +1 @@
+useImageDetails
```
Is there something missing on this slide ?

JonathanGeoffroy committed:
Update src/expert/hooks/README.md

Co-authored-by: Julien Hery <julien_hery@hotmail.fr>

JonathanGeoffroy committed:
Split heavy slide in multiple parts

JonathanGeoffroy committed:
Add solution content

JonathanGeoffroy committed:
exercice -> exercise

JonathanGeoffroy committed:
FIX: Remove useless class

Remove useless class (please see the typo)
Adding the real "with-code" will result in smaller text code, which I personally don't like
